### PR TITLE
Cleaned code

### PR DESCRIPTION
**Pull request by AndreiOla**
**Changed**
I will add comments on each change in the commit.

### CODE REVIEW

File: src/main/java/io/boomerang/service/FlowExecutionServiceImpl.java [hunk 1]
```diff
@@ -284,24 +284,24 @@ private void executeWorkflowAsync(String activityId, final Task start, final Tas
...
+      final Task start, final Task end,
       final Graph<String, DefaultEdge> graph) {
 
     try {
+      final List<String> nodes =
+          GraphProcessor.createOrderedTaskList(graph, start.getTaskId(), end.getTaskId());
       List<Task> nextNodes = this.getTasksDependants(tasks, currentTask);
-      for (Task next : nextNodes) {
-
-        final List<String> nodes =
```
Comment (AndreiOla) [inline src/main/java/io/boomerang/service/FlowExecutionServiceImpl.java:294]: No need to run this so many times in the for loop!

File: src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java [hunk 2]
```diff
@@ -139,22 +139,6 @@ public void createTask(InternalTaskRequest request) {
     taskExecution.setFlowTaskStatus(TaskStatus.inProgress);
     taskExecution = taskActivityService.save(taskExecution);
 
-    // Synchronize status update since `createTask` can be invoked multiple times, depending on the
```
Comment (AndreiOla) [inline src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java:142]: As this code manages Tasks and not Activities, the Status changes for Activities should be managed at a higher level.
This is already done in FlowExectuionServiceImpl->executeWorkflowAsync (line 254)

File: src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java [hunk 3]
```diff
@@ -415,8 +399,6 @@ private void createApprovalNotification(TaskExecutionEntity taskExecution, Task
     taskExecution = taskActivityService.save(taskExecution);
 
     activity.setStatus(TaskStatus.waiting);
-    activityService.saveWorkflowActivity(activity);
```
Comment (AndreiOla) [inline src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java:418]: There is already a save at line 428, so this one is redundant.

File: src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java [hunk 4]
```diff
@@ -498,7 +481,6 @@ public void endTask(InternalTaskResponse request) {
     List<String> keys = new LinkedList<>();
     keys.add(storeId);
 
-    workflowActivity = activityService.findWorkflowActivtyById(activity.getActivityId());
```
Comment (AndreiOla) [inline src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java:501]: I think this Entity is already retrieved at line 461, so no need to get it again.

File: src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java [hunk 5]
```diff
@@ -517,11 +499,8 @@ public void endTask(InternalTaskResponse request) {
     String tokenId = getLock(storeId, keys, 105000);
     LOGGER.debug("[{}] Obtained lock", activityId);
 
-    workflowActivity = activityService.findWorkflowActivtyById(activity.getActivityId());
     updatePendingApprovalStatus(workflowActivity);
 
-    activity.setFlowTaskStatus(request.getStatus());
```
Comment (AndreiOla) [inline src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java:523]: Already set at line 484.

File: src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java [hunk 6]
```diff
@@ -544,7 +523,7 @@ private void updatePendingApprovalStatus(ActivityEntity workflowActivity) {
         ApprovalStatus.submitted);
     boolean existingApprovals = (count > 0);
     workflowActivity.setAwaitingApproval(existingApprovals);
-    activityService.saveWorkflowActivity(workflowActivity);
+    workflowActivity = activityService.saveWorkflowActivity(workflowActivity);
```
Comment (AndreiOla) [inline src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java:526]: Re assign this, as the .saveWorkflowActivity(..) may return a different instance of this entity.

**Review by gchickma — APPROVED** [review]
Looks okay to me

**Pull request merged**